src/node/rpc/memberfrontend.h

```diff
@@ -300,13 +307,50 @@ namespace ccf
         if (proposal->proposer != args.caller_id)
           return jsonrpc::error(
             jsonrpc::StandardErrorCodes::INVALID_REQUEST,
-            "Proposals can only be removed by proposer.");
+            "Proposals can only be removed by proposer");
 
         proposals->remove(proposal_id);
         return jsonrpc::success(true);
       };
       install_with_auto_schema<ProposalAction, bool>(
-        MemberProcs::REMOVAL, removal, Write);
+        MemberProcs::REMOVE, remove, Write);
+
+      auto withdraw = [this](RequestArgs& args) {
+        if (!check_member_status(
+              args.tx, args.caller_id, {MemberStatus::ACTIVE}))
+          return jsonrpc::error(jsonrpc::CCFErrorCodes::INSUFFICIENT_RIGHTS);
+
+        const auto proposal_action = args.params.get<ProposalAction>();
+        const auto proposal_id = proposal_action.id;
+        auto proposals = args.tx.get_view(this->network.proposals);
+        auto proposal = proposals->get(proposal_id);
+
+        if (!proposal)
+          return jsonrpc::error(
+            jsonrpc::StandardErrorCodes::INVALID_PARAMS,
+            "Proposal does not exist");
```

Comment:
Should these errors return the proposal id?

Reply:
Yes, I'll add these